mergify_engine/clients/github_app.py

```diff
@@ -51,36 +51,43 @@
 }
 
 
-class GithubBearerAuth(httpx.Auth):
+class JwtHandler:
     JWT_EXPIRATION = 60
 
     def __init__(self):
         self.jwt = None
         self.jwt_expiration = None
+        self.lock = threading.Lock()
 
-    def get_or_create_jwt(self):
+    def get_or_create(self, force=False):
         now = int(time.time())
+        if force or self.jwt is None or self.jwt_expiration <= now:
```

Comment:
Hm I'd say this needs to be done with the lock to be safe, no?

Reply:
yep, even both to fast and safe.

Reply:
Sorry, I don't understand your reply.

If you don't lock before the `if`, this can happen:

1. Thread 1 evaluates the `if` and it's True
2. Thread 1 locks and starts renewing the JWT
3. Thread 1 changes `self.jwt_expiration` to the future
3. Thread 2 check the if and sees that the expiration time is correct
3. Thread 2 returns the current `self.jwt` which is actually outdated